src/main/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabHelper.java

```diff
@@ -19,18 +23,35 @@ public static GitLabApi apiBuilder(String serverName) {
         GitLabServer server = GitLabServers.get().findServer(serverName);
         if (server != null) {
             PersonalAccessToken credentials = server.getCredentials();
+            String serverUrl = server.getServerUrl();
             if (credentials != null) {
-                return new GitLabApi(server.getServerUrl(), credentials.getToken().getPlainText(), null, getProxyConfig());
+                return new GitLabApi(serverUrl, credentials.getToken().getPlainText(), null, getProxyConfig(serverUrl));
             }
-            return new GitLabApi(server.getServerUrl(), GitLabServer.EMPTY_TOKEN, null, getProxyConfig());
+            return new GitLabApi(serverUrl, GitLabServer.EMPTY_TOKEN, null, getProxyConfig(serverUrl));
         }
         throw new IllegalStateException(
             String.format("No server found with the name: %s", serverName));
     }
 
-    public static Map<String, Object> getProxyConfig () {
+    public static Map<String, Object> getProxyConfig (String serverUrl) {
         ProxyConfiguration proxyConfiguration = Jenkins.get().getProxy();
         if (proxyConfiguration != null) {
+                final URL url;
+                try {
+                    url = new URL(serverUrl);
+                } catch (MalformedURLException e) {
+                    // let it crash somewhere else
+                    return null;
+                }
+                if (!"http".equals(url.getProtocol()) && !"https".equals(url.getProtocol())) {
+                    // non-http(s) URL, proxy won't handle it
+                    return null;
+                }
+                List<Pattern> nonProxyHostPatterns = proxyConfiguration.getNoProxyHostPatterns();
+                if (nonProxyHostPatterns.stream().anyMatch(p -> p.matcher(url.getHost()).matches())) {
```

Comment:
The no_proxy list in Jenkins roughly follows Java syntax (so the exclude would be written as `*.example.com`) and the [getNoProxyHostPatterns](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/ProxyConfiguration.java#L215-L224) converts them into regex patterns to match against hosts, so this will work as expected.